sklearn/decomposition/_nmf.py

```diff
@@ -803,23 +918,30 @@ def _fit_multiplicative_update(X, W, H, beta_loss='frobenius',
     previous_error = error_at_init
 
     H_sum, HHt, XHt = None, None, None
-    for n_iter in range(1, max_iter + 1):
+
+    batches = gen_batches(n_samples, batch_size)
+    batches = itertools.cycle(batches)
+
+    for n_iter, batch in zip(
+        range(iter_offset + 1, iter_offset + max_iter + 1),
+        batches
```

Comment:
Discussions with @ogrisel about the other minibatch estimators lead us agree that ``max_iter`` should control the number of epochs, i.e. number of iterations over the complete dataset. I think it would make sense to try to unify the API across all minibatch estimators. Hence the number of iterations of this loop should be
```
n_steps = (self.max_iter * n_samples) // self._batch_size
```

Then the `n_iter_` attribute should report the number of started epochs and we introduce a new attribute ``n_steps_`` which reports the number of minibatches seen.